src/OrleansServiceBus/Providers/Streams/EventHub/CachePressureMonitors/SlowConsumingPressureMonitor.cs

```diff
@@ -0,0 +1,111 @@
+        {
+        }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="flowControlThreshold"></param>
+        /// <param name="logger"></param>
+        public SlowConsumingPressureMonitor(double flowControlThreshold, Logger logger)
+            : this(flowControlThreshold, DefaultPressureWindowSize, logger)
+        {
+        }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="flowControlThreshold"></param>
+        /// <param name="pressureWindowSzie"></param>
+        /// <param name="logger"></param>
+        public SlowConsumingPressureMonitor(double flowControlThreshold, TimeSpan pressureWindowSzie, Logger logger)
+        {
+            this.FlowControlThreshold = flowControlThreshold;
+            this.logger = logger.GetSubLogger(this.GetType().Name);
+            this.nextCheckedTime = DateTime.MinValue;
+            this.biggestPressureInCurrentWindow = 0;
+            this.wasUnderPressure = false;
+            this.PressureWindowSize = pressureWindowSzie;
+        }
+
+        public void RecordCachePressureContribution(double cachePressureContribution)
+        {
+            if (cachePressureContribution > this.biggestPressureInCurrentWindow)
+                biggestPressureInCurrentWindow = cachePressureContribution;
+        }
+
+        public bool IsUnderPressure(DateTime utcNow)
+        {
+            //if any pressure contribution in current period is bigger than flowControlThreshold
+            //we see the cache is under pressure
+            bool underPressure = this.biggestPressureInCurrentWindow > this.FlowControlThreshold;
+
+            if (underPressure && !this.wasUnderPressure)
+            {
+                //if under pressure, extend the nextCheckedTime, make sure wasUnderPressure is true for a whole window  
+                this.wasUnderPressure = underPressure;
+                this.nextCheckedTime = utcNow + this.PressureWindowSize;
+                logger.Verbose($"Ingesting messages too fast. Throttling message reading. BiggestPressureInCurrentPeriod: {biggestPressureInCurrentWindow}, Threshold: {FlowControlThreshold}");
+                this.biggestPressureInCurrentWindow = 0;
+            }
+
+            if (this.nextCheckedTime < utcNow)
+            {
+                //at the end of each check period, reset biggestPressureInCurrentPeriod
+                this.nextCheckedTime = utcNow + this.PressureWindowSize;
+                this.biggestPressureInCurrentWindow = 0;
+                //if at the end of the window, pressure clears out, log
+                if(this.wasUnderPressure && !underPressure)
+                    logger.Verbose($"Message ingestion is healthy. BiggestPressureInCurrentPeriod: {biggestPressureInCurrentWindow}, Threshold: {FlowControlThreshold}");
+                this.wasUnderPressure = underPressure;
+            }
+
+            return this.wasUnderPressure;
```

Comment:
or if you want something even more conservative, such as :
 
```
public bool IsUnderPressure(DateTime utcNow)
        {
            //if any pressure contribution in current period is bigger than flowControlThreshold
            //we see the cache is under pressure
            bool underPressure = this.biggestPressureInCurrentPeriod > this.FlowControlThreshold;

            if (underPressure)
            {
                //if under pressure, extend the nextCheckedTime, make sure wasUnderPressure is true for a whole window  
                this.wasUnderPressure = underPressure;
                this.nextCheckedTime = utcNow + this.PressureWindowSize;
                logger.Verbose($"Ingesting messages too fast. Throttling message reading. BiggestPressureInCurrentPeriod: {biggestPressureInCurrentPeriod}, Threshold: {FlowControlThreshold}");
                this.biggestPressureInCurrentPeriod = 0;
            }
            else
            {
                if (this.nextCheckedTime < utcNow)
                {
                    //at the end of each check period, reset biggestPressureInCurrentPeriod
                    this.nextCheckedTime = utcNow + this.PressureWindowSize;
                    this.biggestPressureInCurrentPeriod = 0;
                    this.wasUnderPressure = underPressure;
                    if(!this.wasUnderPressure)
                        logger.Verbose($"Message ingestion is healthy. BiggestPressureInCurrentPeriod: {biggestPressureInCurrentPeriod}, Threshold: {FlowControlThreshold}");
                }
            }

            return this.wasUnderPressure;
        }
```

isUnderPressure will be true until slow consumer catch up to above threshold, and be healthy for a whole window